pandapipes/component_models/abstract_models/branch_models.py

```diff
@@ -4,19 +4,17 @@
 
 import numpy as np
 
-from pandapipes.component_models.abstract_models import Component
-from pandapipes.component_models.auxiliaries.derivative_toolbox import calc_der_lambda, calc_lambda
+from pandapipes.component_models.abstract_models.base_component import Component
 from pandapipes.constants import NORMAL_PRESSURE, GRAVITATION_CONSTANT, NORMAL_TEMPERATURE, \
     P_CONVERSION
 from pandapipes.idx_branch import FROM_NODE, TO_NODE, LENGTH, D, TINIT, AREA, K, RHO, ETA, \
     VINIT, RE, LAMBDA, LOAD_VEC_NODES, ALPHA, QEXT, TEXT, LOSS_COEFFICIENT as LC, branch_cols, \
     T_OUT, CP, VINIT_T, FROM_NODE_T, PL, TL, \
     JAC_DERIV_DP, JAC_DERIV_DP1, JAC_DERIV_DT, JAC_DERIV_DT1, JAC_DERIV_DT_NODE, JAC_DERIV_DV, \
-    JAC_DERIV_DV_NODE, \
-    LOAD_VEC_BRANCHES, LOAD_VEC_BRANCHES_T, LOAD_VEC_NODES_T, ELEMENT_IDX
+    JAC_DERIV_DV_NODE, LOAD_VEC_BRANCHES, LOAD_VEC_BRANCHES_T, LOAD_VEC_NODES_T
 from pandapipes.idx_node import PINIT, HEIGHT, TINIT as TINIT_NODE, PAMB
-from pandapipes.internals_toolbox import _sum_by_group, select_from_pit
-from pandapipes.pipeflow_setup import get_table_number, get_lookup
+from pandapipes.pf.derivative_calculation import calc_lambda, calc_der_lambda
+from pandapipes.pf.pipeflow_setup import get_table_number, get_lookup
 from pandapipes.properties.fluids import get_fluid
 
```

Comment:
I removed the function for hydraulic calculation. The one for thermal calculation is still needed. We had discussed that it might be good to let some components implement their own derivative behavior, but this requires some changes elsewhere.